docs/using-docker.md

```diff
@@ -0,0 +1,38 @@
+# Using Docker for job execution
+
+## Overview 
+Turbinia can support Docker by allowing users to configure jobs to execute through a Docker container. This eliminates the need to install all the necessary dependencies and/or external programs on the host machine and instead execute the job through a container that has all the necessary dependencies installed. Please note that Turbinia does not provide the Docker images necessary for each job and they will need to either be created or pulled from a container registry. 
```

Comment:
I think we should clarify here what exactly is running in the container since there's a difference between running the Job vs. the Task (i.e. the actual python Task code) vs. the binaries executed by the Task.  I mention this because separately we have talked about running the entire Task or Worker in a container.